pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -564,4 +570,18 @@ in
             scheme-medium scheme-minimal scheme-small scheme-tetex;
         }
     );
-  }
+  };
+
+  texlive = makeScopeWithSplicing' {
+    f = makeTeXLive;
+    otherSplices = generateSplicesForMkScope "texlive";
+  };
+in
+# backward compatibility layer: expose packages directly under texlive, but not as derivations
+# this cannot be done from within the scope
+(builtins.mapAttrs
+  # flag if manpages are present
+  (n: p: let pWithMan = p // lib.optionalAttrs (p ? man) { texdoc = p.texdoc // { hasManpages = true; }; }; in {
+    pkgs = builtins.concatMap (n: lib.optional (p ? ${n}) p.${n})  [ "tex" "texdoc" "texsource" "tlpkg" "out" ];
+  })
+  texlive.__pkgs) // texlive
```

Comment:
The compromise here is that you cannot use `inherit (texlive.overrideScope ...) package`, so this must come after `buildEnv`.

Reply:
...or we do @apfelkuchen6's `applyOverScope` trick:
https://github.com/NixOS/nixpkgs/pull/250626/commits/13e78086cee9dbfb1d9dcb531cbbaf899987c52c#diff-78dd1e8609118bb30c483b837f017110017218a9b7c388887a9c0ed0aa823f36R162-R176